camus-sweeper/src/main/java/com/linkedin/camus/sweeper/CamusSweeper.java

```diff
@@ -274,10 +277,20 @@ private void runCollectorForTopicDir(FileSystem fs, String topic, Path topicSour
 
     List<Properties> jobPropsList = planner.createSweeperJobProps(topic, topicSourceDir, topicDestDir, fs);
 
-    for (Properties jobProps : jobPropsList){
+    for (Properties jobProps : jobPropsList) {
       tasksToComplete.add(runCollector(jobProps, topic));
     }
 
+    for (Future<?> task : tasksToComplete) {
+      try {
+        task.get();
+      } catch (InterruptedException e) {
+        throw e;
+      } catch (ExecutionException e) {
+        throw e;
+      }
+    }
+
     log.info("Finishing processing for topic " + topic);
```

Comment:
Hmm yes good catch also. I'll see if there is a way to make it bubble up exception somehow else.